gazebo_ros/include/gazebo_ros/utils_visibility_control.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2020 Microsoft Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* This header must be included by all rclcpp headers which declare symbols
+ * which are defined in the rclcpp library. When not building the rclcpp
+ * library, i.e. when using the headers in other package's code, the contents
+ * of this header change the visibility of certain symbols which the rclcpp
+ * library cannot have, but the consuming code must have inorder to link.
+ */
+
+#ifndef GAZEBO_ROS__UTILS_VISIBILITY_CONTROL_H_
+#define GAZEBO_ROS__UTILS_VISIBILITY_CONTROL_H_
+
+// This logic was borrowed (then namespaced) from the examples on the gcc wiki:
+//     https://gcc.gnu.org/wiki/Visibility
+
+#if defined _WIN32 || defined __CYGWIN__
+  #ifdef __GNUC__
+    #define GAZEBO_ROS_UTILS_EXPORT __attribute__ ((dllexport))
```

Comment:
On Gazebo and Ignition, we've been [using](https://bitbucket.org/osrf/gazebo/src/5f48d294547f2eccf368de73080470290dd6a3b1/gazebo/util/system.hh#lines-42) the `VISIBLE` suffix. Unless there's a good reason to use `EXPORT`, I'd suggest sticking to the existing pattern.

Reply:
I don't have strong opinion on the naming. The `visibility_control.h` is largely taken from the examples I can see from [ROS 2 GitHub org](https://github.com/search?q=org%3Aros2+filename%3Avisibility_control.h&type=Code). If you prefer to use `VISIBLE` suffix convention like Gazebo, then I can rename it.

Reply:
Thanks for the links, following ROS 2 convention sounds good to me.

Reply:
I agree here.